pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -199,6 +213,19 @@ func getVirtualMachineNetworks(virtualMachine *kubevirt.VirtualMachine) []kubevi
 	return virtualMachine.Spec.Template.Spec.Networks
 }
 
+func checkVmForInterfaceDuplication(virtualMachine *kubevirt.VirtualMachine) bool {
+	vmInterfaces := getVirtualMachineInterfaces(virtualMachine)
+
+	tmpInterfaceMap := map[string]string{}
+	for _, vmInterface := range vmInterfaces {
+		if _, exists := tmpInterfaceMap[vmInterface.Name]; exists {
+			return false
+		}
+		tmpInterfaceMap[vmInterface.Name] = ""
+	}
+	return true
+}
```

Comment:
```suggestion
func checkVmForInterfaceDuplication(virtualMachine *kubevirt.VirtualMachine) error {
	vmInterfaces := getVirtualMachineInterfaces(virtualMachine)

	tmpInterfaceMap := map[string]string{}
	for _, vmInterface := range vmInterfaces {
		if _, exists := tmpInterfaceMap[vmInterface.Name]; exists {
			return fmt.Errorf("Failed to mutate virtual machine: every interface must have a unique name")
		}
		tmpInterfaceMap[vmInterface.Name] = ""
	}
	return nil
}
```

This way on the two places we call it we don't have to duplicate the error

```golang
if err := checkVmForInterfaceDuplication(virtualMachine) ; err != nil {
    return err
}
		}
```

Reply:
Done